frontend/src/components/AddClassSearchModal.tsx

```diff
@@ -0,0 +1,308 @@
+  height: 200px;
+  width: 100%;
+  margin-top: 20px;
+  margin-bottom: 10px;
+  border: 1px solid red;
+  border-radius: 10px;
+  overflow-y: scroll;
+  font-family: Roboto;
+  font-style: normal;
+  font-weight: normal;
+`;
+
+const SearchResultContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  padding: 10px;
+  margin-left: 15px;
+  margin-right: 30px;
+`;
+
+const NoResultContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  padding: 10px;
+  margin-left: 15px;
+  margin-right: 30px;
+`;
+
+const Loading = styled.div`
+  font-size: 10px;
+  line-height: 10px;
+  margin-top: 10px;
+  margin-bottom: 5px;
+  margin-left: 10px;
+  margin-right: 10px;
+`;
+
+const AddedClassesContainer = styled.div``;
+
+const ResultInfoContainer = styled.div``;
+
+const AddClassButton = styled.div`
+  padding: 5px;
+  background: pink;
+  width: 18px;
+  height: 18px;
+  -moz-border-radius: 50px;
+  -webkit-border-radius: 50px;
+  border-radius: 50px;
+  align-items: center;
+  &:hover {
+    background: red;
+  }
+`;
+
+const AddClassError = styled.div`
+  background: pink;
+  color: white;
+  width: 18px;
+  height: 18px;
+  padding: 5px;
+  font-size: 16px;
+  font-weight: 800;
+  -moz-border-radius: 50px;
+  -webkit-border-radius: 50px;
+  border-radius: 50px;
+  text-align: center;
+`;
+
+const SubjectId = styled.div`
+  font-size: 10px;
+  line-height: 21px;
+`;
+
+const CourseName = styled.div`
+  font-size: 15px;
+  line-height: 21px;
+`;
+
+const EMPTY_COURSE_LIST: ScheduleCourse[] = [];
+
+interface SearchResultProps {
+  course: ScheduleCourse;
+}
+
+export const AddClassSearchModal: React.FC<
+  AddClassSearchModalProps
+> = props => {
+  const [searchQuery, setSearchQuery] = useState("");
+  const [isLoading, setIsLoading] = useState(false);
+  const [searchedCourses, setSearchedCourses] = useState(EMPTY_COURSE_LIST);
+  const [selectedCourses, setSelectedCourses] = useState(EMPTY_COURSE_LIST);
+  const { schedule } = useSelector((state: AppState) => ({
+    schedule: safelyGetActivePlanScheduleFromState(state),
+  }));
+
+  const SearchResult = (props: SearchResultProps) => {
+    let showCourseInSchedyleError = false;
+    if (schedule != null) {
+      showCourseInSchedyleError = isCourseInSchedule(props.course, schedule);
+    }
+    return (
+      <SearchResultContainer
+        key={props.course.subject + " " + props.course.classId}
+      >
+        <ResultInfoContainer>
+          <CourseName>{props.course.name}</CourseName>
+          <SubjectId>
+            {props.course.subject + " " + props.course.classId}
+          </SubjectId>
+        </ResultInfoContainer>
+        {showCourseInSchedyleError ? (
+          <Tooltip title="Course already in schedule" aria-label="add">
+            <AddClassError>!</AddClassError>
+          </Tooltip>
+        ) : (
+          <AddClassButton
+            onClick={async () => {
```

Comment:
Can we pull this out into a function in the component as opposed to in the render?